File: Lua/XSignInstallCocos2dxLua/script/xsigninstall.lua

local targetPlatform = cc.Application:getInstance():getTargetPlatform()

local xsigninstall = class("xsigninstall")

local activityClassName = cc.PLATFORM_OS_ANDROID == targetPlatform and "io/xsigninstall/cocos2dx/XSignInstallInterface" or "XSignInstallBridge"

function xsigninstall:reportRegister()
	print("call reportRegister start")
	if (cc.PLATFORM_OS_ANDROID == targetPlatform) then
        local luaj = require "cocos.cocos2d.luaj"
		local args = {}
		local signs = "()V"
		local ok,ret = luaj.callStaticMethod(activityClassName, "reportRegister", args, signs)
		if not ok then
            print("call reportRegister fail" .. ret)
		end
	end
    if (cc.PLATFORM_OS_IPHONE == targetPlatform) or (cc.PLATFORM_OS_IPAD == targetPlatform) then
        local luaoc = require "cocos.cocos2d.luaoc"
        local ok, ret = luaoc.callStaticMethod(activityClassName, "reportRegister")
        if not ok then
            print("luaoc reportRegister error:" .. ret)
        end
    end
end

function xsigninstall:getInstall(callback)
    print("call getInstall start")
	if (cc.PLATFORM_OS_ANDROID == targetPlatform) then
        local luaj = require "cocos.cocos2d.luaj"
		local args = {callback}
		local signs = "(I)V"
		local ok,ret = luaj.callStaticMethod(activityClassName, "getInstall", args, signs)
		if not ok then
			print("call getInstall fail" .. ret)
		end
	end
    if (cc.PLATFORM_OS_IPHONE == targetPlatform) or (cc.PLATFORM_OS_IPAD == targetPlatform) then
        local luaoc = require "cocos.cocos2d.luaoc"
        local args = {callback = callback}
        local ok, ret = luaoc.callStaticMethod(activityClassName, "getInstall", args)
        if not ok then
            print("luaoc getInstall error:" .. ret)
        end
    end
end

function xsigninstall:registerWakeupCallback(callback)
	print("call registerWakeupCallback start")
	if (cc.PLATFORM_OS_ANDROID == targetPlatform) then
        local luaj = require "cocos.cocos2d.luaj"
		local args = {callback}
		local signs = "(I)V"
		local ok,ret = luaj.callStaticMethod(activityClassName, "registerWakeupCallback", args, signs)
		if not ok then
			print("call registerWakeupCallback fail" .. ret)
		end
	end
    if (cc.PLATFORM_OS_IPHONE == targetPlatform) or (cc.PLATFORM_OS_IPAD == targetPlatform) then
        local luaoc = require "cocos.cocos2d.luaoc"
        local args = {callback = callback}
        local ok, ret = luaoc.callStaticMethod(activityClassName, "registerWakeupCallback", args)
        if not ok then
            print("luaoc registerWakeupCallback error:" .. ret)
        end
    end
end

return xsigninstall

File: Creator/XSignInstallCocosCreator/script/XSignInstall.ts
export default class XSignInstall {
    private static activityClassName = cc.sys.OS_ANDROID == cc.sys.os ? "io/xsigninstall/cocos2dx/XSignInstallInterface" : "XSignInstallBridge";
    private static _wakeupCallback(appData){

    }

    private static _installCallback(appData){

    }

    public static getInstall(callback) {
        this._installCallback = callback;
        if (cc.sys.OS_ANDROID == cc.sys.os) {
            jsb.reflection.callStaticMethod(this.activityClassName, "getInstall", "()V");
        } else if(cc.sys.OS_IOS == cc.sys.os){
            jsb.reflection.callStaticMethod(this.activityClassName, "getInstall");
        }
    }
    public static registerWakeupCallback(callback) {
        this._wakeupCallback = callback;
        if (cc.sys.OS_ANDROID == cc.sys.os) {
            jsb.reflection.callStaticMethod(this.activityClassName, "registerWakeupCallback", "()V");
        } else if(cc.sys.OS_IOS == cc.sys.os){
            jsb.reflection.callStaticMethod(this.activityClassName, "registerWakeupCallback");
        }
    }

    public static reportRegister() {
        if (cc.sys.OS_ANDROID == cc.sys.os) {
            jsb.reflection.callStaticMethod(this.activityClassName, "reportRegister", "()V");
        } else if(cc.sys.OS_IOS == cc.sys.os){
            jsb.reflection.callStaticMethod(this.activityClassName, "reportRegister");
        }
    }

    public static installCallback(data) {
        cc.log("安装参数：" + JSON.stringify(data));
        this._installCallback(data);
    }

    public static wakeupCallback(data) {
        cc.log("拉起参数：" + JSON.stringify(data));
        this._wakeupCallback(data);
    }
};
globalThis.XSignInstall = XSignInstall;